Feature: Add GitHub action workflow

maxhoheiser opened the pull request:
Add a GitHub action workflow, that installs required dependencies, starts a Localstack container, deploys the infrastructure and runs the pytest

maxhoheiser committed:
feat: add github workflow

joe4dev commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,95 @@
+  schedule:
+    # “At 00:00 on Sunday.”
+    - cron: "0 0 * * 0"
+  workflow_dispatch:
```
not needed?

maxhoheiser replied:
deleted in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/c7bb4b90b9906faeceeb60e07b5bbc03ab7161d0

joe4dev commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,95 @@
+            LOCALSTACK_API_KEY: ${{ secrets.LOCALSTACK_API_KEY }}
+
+      - name: Deploy using CDK
+        run: |
```
Consider invoking the make targets here

joe4dev replied:
but that makes it a bit trickier to control the exit codes. Chaining with `&& \` instead of newlines would do it

joe4dev commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,95 @@
+            cdklocal deploy KinesisFirehoseRedshiftStack2 --require-approval never
+
+      - name: List the resources
+        run: |
```
e.g., `make list_resources`

maxhoheiser replied:
fixed in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/d590f1498acf26c3ccb669aaac0740957b5d4355

joe4dev commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,95 @@
+            awslocal redshift describe-clusters
+
+      - name: Run pytests
+        run: |
```
`make test`

maxhoheiser replied:
fixed in: https://github.com/localstack-samples/sample-cdk-kinesis-firehose-redshift/pull/3/commits/50721fca533bcfcb41427357e58866090cc815ec

joe4dev reviewed (changes requested):
It looks like the workflow is failing with the error `TypeError: type of argument db_name must be str; got NoneType instead`

Is this waiting for a dependent PR?

maxhoheiser committed:
feat: remove schedule to run the action

maxhoheiser committed:
feat: use make command to deploy

maxhoheiser committed:
feat: use make command to run test

maxhoheiser committed:
feat: use make command to list resources

maxhoheiser committed:
feat: add specific environment with variables and secrets to avoid checking in .env file

maxhoheiser committed:
feat: add explanation to prepare github environment variables

HarshCasper committed:
try fixing the CI

maxhoheiser committed:
feat: using .env file in github action

Even if this exposes secrets and config variables, since they should only be mock variables used for local testing it is not a security risk. For production always use a github actions environment with secrets and variables.
